pkg/proxy/proxier.go

```diff
@@ -565,12 +573,35 @@ func (proxier *Proxier) openPortal(service ServicePortName, info *serviceInfo) e
 	if err != nil {
 		return err
 	}
+
+	publicIpCount := 0
 	for _, publicIP := range info.publicIPs {
-		err = proxier.openOnePortal(net.ParseIP(publicIP), info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
+		ip := net.ParseIP(publicIP)
+		if ip == nil {
+			glog.V(2).Info("Skipping non-address public endpoint: %s", publicIP)
+			continue
+		}
+		publicIpCount++
+		err = proxier.openOnePortal(ip, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
 		if err != nil {
 			return err
 		}
 	}
+
+	// We have an external load balancer, but no specific IPs.
+	// This indicates that the load balancer is acting as TCP relay, targeting the machine's IP.
+	// We also treat DNS names the same way, because DNS names are used when we have multiple
+	// addresses for the external load balancer (e.g. AWS ELB)
+	// TODO: This is basically a hack.  If we create load balancers asynchronously it'll likely break.
+	// We probably need to specify the cloud as an argument to kube-proxy.
+	if info.externalLoadBalancer && publicIpCount == 0 {
+		glog.V(2).Info("Forwarding node public ip for service:", service.Name)
+		err = proxier.openOnePortal(proxier.hostIP, info.portalPort, info.protocol, proxier.listenIP, info.proxyPort, service)
```

Comment:
We need to make sure that the port doesn't collide with any of the kube components running on the node, i.e. the kubelet and the kube-proxy itself. I don't think the proxy should be an issue since it typically only listens on localhost, but you may want to make sure this won't interfere with either.